src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs

```diff
@@ -57,6 +80,8 @@ internal TestAssetInfo(DirectoryInfo root, string assetName)
                 _dataDirectory
             };
 
+            RemoveDataDirectoryIfAssetChanged();
```

Comment:
agreed on the practice, but in my experience with TestAssets the biggest pain point has been during the adjustment of an asset stale content gets copied over. I wanted to be sure that this is always hit to keep us from being frustrated. Do you have a proposal?